lib/ansible/playbook/play_context.py

```diff
@@ -424,7 +423,7 @@ def detect_su_prompt(data):
                     return bool(SU_PROMPT_LOCALIZATIONS_RE.match(data))
 
                 prompt = detect_su_prompt
-                becomecmd = '%s %s %s -c "%s -c %s"' % (exe, flags, self.become_user, executable, success_cmd)
+                becomecmd = '%s %s %s -c %s' % (exe, flags, self.become_user, pipes.quote("%s -c %s" % (shell, success_cmd)))
```

Comment:
I tried this before and it broke in some configurations, I need to look back to see exactly which